src/optimizer/pullup/pullup_projection.cpp

```diff
@@ -0,0 +1,113 @@
+#include "duckdb/optimizer/filter_pullup.hpp"
+#include "duckdb/planner/expression/bound_columnref_expression.hpp"
+#include "duckdb/planner/expression_iterator.hpp"
+#include "duckdb/planner/operator/logical_empty_result.hpp"
+#include "duckdb/planner/operator/logical_projection.hpp"
+#include "duckdb/planner/expression/bound_comparison_expression.hpp"
+
+namespace duckdb {
+using namespace std;
+
+static Expression *GetColumnRefExpression(Expression &expr) {
+    if (expr.type == ExpressionType::BOUND_COLUMN_REF) {
+        return &expr;
+    }
+    ExpressionIterator::EnumerateChildren(expr, [&](Expression &child) { return GetColumnRefExpression(child); });
+    return &expr;
+}
+
+static bool GenerateBinding(LogicalProjection &proj, BoundColumnRefExpression &colref, ColumnBinding &binding) {
+    D_ASSERT(colref.depth == 0);
+    int column_index = -1;
+    // find the corresponding column index in the projection
+    for(idx_t proj_idx=0; proj_idx < proj.expressions.size(); proj_idx++) {
+        auto proj_colref = GetColumnRefExpression(*proj.expressions[proj_idx]);
```

Comment:
It seems that this can be replaced with `auto proj_colref = proj.expressions[proj_idx].get()` and the function `GetColumnRefExpression` can be removed.